scilpy/tractograms/tractogram_operations.py

```diff
@@ -898,6 +960,355 @@ def split_sft_randomly_per_cluster(orig_sft, chunk_sizes, seed, thresholds):
     return final_sfts
 
 
+def subsample_streamlines_alter(sft, min_dice=0.90, epsilon=0.01,
+                                baseline_sft=None):
+    """
+    Function to subsample streamlines based on a dice similarity metric.
+    The function will keep removing streamlines until the dice similarity
+    between the original and the subsampled tractogram is close to min_dice.
+
+    Parameters
+    ----------
+    sft: StatefulTractogram
+        The tractogram to subsample.
+    min_dice: float
+        The minimum dice similarity to reach before stopping the subsampling.
+    epsilon: float
+        Stopping criteria for convergence. The maximum difference between the
+        dice similarity and min_dice.
+    baseline_sft: StatefulTractogram
+        The tractogram to use as a reference for the dice similarity. If None,
+        the original tractogram will be used.
+
+    Returns
+    -------
+    new_sft: StatefulTractogram
+        The tractogram with a subset of streamlines in the same space as the
+        input tractogram.
+    """
+    # Import in function to avoid circular import error
+    from scilpy.tractanalysis.reproducibility_measures import compute_dice_voxel
```

Comment:
This means that our management is not perfect yet.

`tractanalysis.reproducibility_measures.compute_dice_voxel` seems to be usable with any map. So it is not necessarily a tractanalysis method. Maybe it could go in `image.volume_operations` (or in volume_math if we also want to offer this option in scil_volume_math). Or in stats.something but I don't like it as much.

Reply:
I believe that it makes more sense to have the `compuet_dice_voxel` and `compute_dice_streamline`, and the bundle adjacency voxel and bundle adjacency voxel next to each other than splitting them because of the module name 'tractanalysis'.

Like `min_dist_to_centroid` works with point cloud, not just streamline. or `compute_f1_score` or `compute_f1_overlap_overreach` works with mask too. I believe it is alright since almost everything we do in scilpy is tract-oriented, even if some functions could be used differently sometimes.